Public/Src/App/Bxl/BuildXLApp.cs

```diff
@@ -979,6 +981,47 @@ private AppResult RunWithLoggingScope(Func<PerformanceMeasurement, AppResult> ru
             return result;
         }
 
+        /// <summary>
+        /// Computes session identifier which allows easier searching in Kusto for 
+        /// builds based traits: Cloudbuild BuildId (i.e. RelatedActivityId), ExecutionEnvironment, Distributed build role
+        /// 
+        /// Search for masters: '| where sessionId has "0001-FFFF"'
+        /// Search for workers: '| where sessionId has "0002-FFFF"'
+        /// Search for office metabuild: '| where sessionId has "FFFF-0F"'
+        /// </summary>
+        private Guid ComputeSessionId(Guid relatedActivityId)
+        {
+            var bytes = relatedActivityId.ToByteArray();
+            var executionEnvironment = m_configuration.Logging.Environment;
+            var distributedBuildRole = m_configuration.Distribution.BuildRole;
+            var inCloudBuild = m_configuration.InCloudBuild();
+
+            // SessionId:
+            // 00-03: 00-03 from random guid
+            var randomBytes = Guid.NewGuid().ToByteArray();
+            for (int i = 0; i <= 3; i++)
+            {
+                bytes[i] = randomBytes[i];
+            }
+
+            // 04-05: BuildRole
+            bytes[4] = 0;
+            bytes[5] = (byte)distributedBuildRole;
+
+            // 06-07: InCloudBuild = FFFF, !InCloudBuild = 0000
+            var inCloudBuildSpecifier = inCloudBuild ? byte.MaxValue : (byte)0;
+            bytes[6] = inCloudBuildSpecifier;
+            bytes[7] = inCloudBuildSpecifier;
+
+            // 08-09: executionEnvironment
+            bytes[8] = (byte)(((int)executionEnvironment >> 8) & 0xFF);
+            bytes[9] = (byte)((int)executionEnvironment & 0xFF);
+

```

Comment:
double checking low legit it is to steal this many bits:
[https://en.wikipedia.org/wiki/Universally_unique_identifier#Collisions](https://en.wikipedia.org/wiki/Universally_unique_identifier#Collisions) 

Updating that from using 122 bits down to 98 yields  
1/2 + (1/4 + 2 + ln(2) * 2^98)^.5 = 4x10^14 guids needing to be generated for probability of collision being 50%. So it seems fine to me at our scale